src/psyclone/dynamo0p3.py

```diff
@@ -3092,6 +3074,100 @@ class DynProxies(LFRicCollection):
     generation since Kernels know nothing about proxies.
 
     '''
+    def __init__(self, node):
+        super().__init__(node)
```

Comment:
It's all a bit confusing. A proxy contains a pointer to the data array inside the Field object but that data array might itself be a pointer I think. All of the dereferencing we do in the PSy layer is from a Proxy because a Field deliberately keeps things private. It seemed to be just the arrays that caused us problems - scalars are fine, as you say. (Probably because they get copied by value?) There are some other pointers (related to maps and things) but they are not inside derived types and so they already work OK. So, to answer your first question, I think it was the fact that it was an array (pointer or not) inside a derived type that caused problems for the data movement. As soon as the compiler saw that the array was in a derived type, it seemed to examine all the other members of the derived type and then got upset if they happened to point to bits of memory shared with other types.